### PR TITLE
CQLHS-497: fix type issues

### PR DESCRIPTION
**Pull request by charkour**
- Updated the Style Preview to use real data.
- Removed the placeholders
- Updated goals widget to pass a goal object
- Removed extra parameters
- Fixed all type errors

TODO: Need to smoke test.

### CODE REVIEW

File: ClientApp/src/components/pages/StylePreviewPage/StylePreviewPage.tsx [hunk 1]
```diff
@@ -137,7 +158,7 @@ export const StylePreviewPage: React.FC = (): JSX.Element => {
               getItemValue={(item: Employee): string => {
                 return item.firstName || "";
               }}
-              items={placeholderEmployeeOverviews}
+              items={employees}
```
Comment (jonathanchaffer) [inline ClientApp/src/components/pages/StylePreviewPage/StylePreviewPage.tsx:161]: Not sure the style preview page should use actual data, but it's not a huge deal.

File: ClientApp/src/services/projectService.ts [hunk 2]
```diff
@@ -184,26 +184,27 @@ export const sendProjectEvaluation = async (
...
           };
         },
       ) || [],
-    links: project.links?.filter((link: LinkSnippet): boolean => {
-      return link.url !== undefined;
-    }),
-    managerId: manager?.employeeId,
+    links:
+      project.links?.filter((link: LinkSnippet): boolean => {
+        return link.url !== undefined;
```
Comment (jonathanchaffer) [inline ClientApp/src/services/projectService.ts:201]:
```suggestion
        return link.url !== undefined && link.url !== "";
```
This should fix the validation error

File: ClientApp/src/components/pages/ProjectEditPage/ProjectEditPage.tsx [hunk 3]
```diff
@@ -102,18 +103,23 @@ export const ProjectEditPage: React.FC = (): JSX.Element => {
...
+      evaluationStatus: -1,
+      evaluations: [],
+      lastModifiedBy: fakeNoNameEmployeeSnippet,
       links: [
-        { linkId: undefined, name: "Jira Project", url: undefined },
-        { linkId: undefined, name: "Confluence Space", url: undefined },
-        { linkId: undefined, name: "Project Retro", url: undefined },
+        { linkId: -1, name: "Jira Project", url: "" },
+        { linkId: -1, name: "Confluence Space", url: "" },
+        { linkId: -1, name: "Project Retro", url: "" },
```
Comment (jonathanchaffer) [inline ClientApp/src/components/pages/ProjectEditPage/ProjectEditPage.tsx:116]:
```suggestion
        { linkId: 0, name: "Jira Project", url: "" },
        { linkId: 0, name: "Confluence Space", url: "" },
        { linkId: 0, name: "Project Retro", url: "" },
```
To fix 500 errors. If the entity doesn't exist yet, its id should be 0.

File: ClientApp/src/components/pages/ProjectEditPage/ProjectEditPage.tsx [hunk 4]
```diff
@@ -102,18 +103,23 @@ export const ProjectEditPage: React.FC = (): JSX.Element => {
...
       ],
+      manager: fakeNoNameEmployeeSnippet,
       notes: undefined,
-      projectName: undefined,
-      startDate: undefined,
+      projectId: -1,
```
Comment (jonathanchaffer) [inline ClientApp/src/components/pages/ProjectEditPage/ProjectEditPage.tsx:120]:
```suggestion
      projectId: 0,
```

File: ClientApp/src/components/pages/StylePreviewPage/StylePreviewPage.tsx (hunk 1 above)
Comment (charkour) [reply ClientApp/src/components/pages/StylePreviewPage/StylePreviewPage.tsx:161]: Yeah, it was either use real data or add 7 required fields to the placeholder data.

File: ClientApp/src/components/pages/GoalDetailPage/UpdatesWidget.tsx [hunk 5]
```diff
@@ -195,52 +197,55 @@ export const UpdatesWidget: React.FC<UpdatesWidgetProps> = ({
...
-                              {update.description || <em>No Notes</em>}
-                            </span>
-                          </Card.Body>
-                        </Accordion.Collapse>
-                      </>
-                    );
-                  },
-                )}
+              {updates ||
+                []
```
Comment (jonathanchaffer) [inline ClientApp/src/components/pages/GoalDetailPage/UpdatesWidget.tsx:203]:
```suggestion
              {(updates || [])
```
To fix runtime error

Comment (charkour) [reply ClientApp/src/components/pages/GoalDetailPage/UpdatesWidget.tsx:203]: Good call. Thanks

File: ClientApp/src/services/goalsService.ts [hunk 6]
```diff
@@ -164,29 +175,30 @@ export const toggleTaskCompletion = async (
 ): Promise<void> => {
   const response = await apiInstance(GoalsApi).apiGoalsSaveMilestonesPost([
     {
-      content: task?.content,
+      content: task?.content || "",
       dueDate: task?.dueDate ? moment(task.dueDate).toISOString() : undefined,
       goal: goal as Goal,
```
Comment (jonathanchaffer) [inline ClientApp/src/services/goalsService.ts:180]:
```suggestion

```
To fix a 400 error. (`Milestone` shouldn't have a `goal` field, infinite nesting)

File: ClientApp/src/components/pages/StylePreviewPage/StylePreviewPage.tsx (hunk 1 above)
Comment (jonathanchaffer) [reply ClientApp/src/components/pages/StylePreviewPage/StylePreviewPage.tsx:161]: ah, good choice then

File: ClientApp/src/services/projectService.ts [hunk 7]
```diff
@@ -184,26 +184,27 @@ export const sendProjectEvaluation = async (
...
+      }) || [],
+    managerId: manager?.employeeId || -1,
     notes: project.notes,
     projectId: project.projectId,
```
Comment (jonathanchaffer) [inline ClientApp/src/services/projectService.ts:205]:
```suggestion
```
To fix a 500 error

Comment (charkour) [reply ClientApp/src/services/projectService.ts:205]: Good, but you'll have to check to see if the project is being updated.

Comment (jonathanchaffer) [reply ClientApp/src/services/projectService.ts]: Good call. Try this:
`projectId: project.projectId || null,`

File: ClientApp/src/components/reuseables/ProjectDetail/ProjectDetail.tsx [hunk 8]
```diff
@@ -128,7 +128,6 @@ export const ProjectDetail: React.FC<ProjectDetailProps> = ({
                         );
                       },
                     )}
-                    ,
```
Comment (charkour) [inline ClientApp/src/components/reuseables/ProjectDetail/ProjectDetail.tsx:131]: haha how did you find that?

Comment (jonathanchaffer) [reply ClientApp/src/components/reuseables/ProjectDetail/ProjectDetail.tsx:131]: Thought it was weird how the last list item was larger than the rest, so I did some investigating 🔎

File: ClientApp/src/services/goalsService.ts (hunk 6 above)
Comment (charkour) [reply ClientApp/src/services/goalsService.ts:180]: This doesn't seem to be resolving the 400

**Pull request merged**